crates/runestick/src/generator.rs

```diff
@@ -20,14 +33,6 @@ impl Generator {
         }
     }
 
-    /// Get the next value produced by this stream.
-    pub fn next(&mut self) -> Result<Option<Value>, VmError> {
```

Comment:
This would be used in modules to iterate over a generator, and they usually have a signature which includes `Result<T, VmError>`. Moving this impl into `Iterator` would just result in every call having to "untranspose" for no benefit.

I'd suggest adding an ignore for the lint, or renaming the function. But `next` is good name, so I'm leaning on ignoring the lint.